tracing/config/tracing.go

```diff
@@ -21,55 +21,99 @@ import (
 	"fmt"
 	"strconv"
 
+	"cloud.google.com/go/compute/metadata"
 	corev1 "k8s.io/api/core/v1"
 )
 
 const (
 	// ConfigName is the name of the configmap
 	ConfigName = "config-tracing"
 
-	enableKey         = "enable"
-	zipkinEndpointKey = "zipkin-endpoint"
-	debugKey          = "debug"
-	sampleRateKey     = "sample-rate"
+	enableKey               = "enable"
+	backendKey              = "backend"
+	zipkinEndpointKey       = "zipkin-endpoint"
+	debugKey                = "debug"
+	sampleRateKey           = "sample-rate"
+	stackdriverProjectIDKey = "stackdriver-project-id"
+)
+
+// BackendType specifies the backend to use for tracing
+type BackendType string
+
+const (
+	// None is used for no backend.
+	None BackendType = "none"
+	// Stackdriver is used for Stackdriver backend.
+	Stackdriver BackendType = "stackdriver"
+	// Zipkin is used for Zipkin backend.
+	Zipkin BackendType = "zipkin"
 )
 
 // Config holds the configuration for tracers
 type Config struct {
-	Enable         bool
-	ZipkinEndpoint string
-	Debug          bool
-	SampleRate     float64
+	Backend              BackendType
+	ZipkinEndpoint       string
+	StackdriverProjectID string
+
+	Debug      bool
+	SampleRate float64
 }
 
 // Equals returns true if two Configs are identical
 func (cfg *Config) Equals(other *Config) bool {
-	return other.Enable == cfg.Enable && other.ZipkinEndpoint == cfg.ZipkinEndpoint && other.Debug == cfg.Debug && other.SampleRate == cfg.SampleRate
+	return other.Backend == cfg.Backend &&
```

Comment:
why not just return `cmp.equals` or `reflect.DeepEquals`?

Reply:
I had no idea what this was being used for, so I didn't want to change it too much.  I made it `reflect.DeepEquals`.  #yolo.